CMakeLists.txt

```diff
@@ -252,13 +252,17 @@ endif()
 # Set native march/mcpu
 if(WITH_NATIVE_INSTRUCTIONS)
     if(CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
-        if(BASEARCH_PPC_FOUND)
-            set(NATIVEFLAG "-mcpu=native")
-        else()
+        check_c_compiler_flag(-march=native HAVE_MARCH_NATIVE)
+        check_c_compiler_flag(-mcpu=native HAVE_MCPU_NATIVE)
+        if(HAVE_MARCH_NATIVE)
             set(NATIVEFLAG "-march=native")
+        elseif(HAVE_MCPU_NATIVE)
+            set(NATIVEFLAG "-mcpu=native")
+        endif()
+        if(NATIVEFLAG)
+            # Apply flag to all source files and compilation checks
+            add_compile_options(${NATIVEFLAG})
         endif()
```

Comment:
Perhaps have a similar message as below if compiler does not support native.
Might be best to make it a warning so it is more visible. Or perhaps an error, it is not likely native will be used by distributions, so should not break stuff. Either is good.

Reply:
I thought about adding warning... But then I also thought about reordering the code that it would just fall through and avoid duplication of the warning message... I wasn't sure which solution would be cleaner...